request.go

```diff
@@ -5,147 +5,154 @@ import (
 	"net/http"
 )
 
-// Request is a GraphQL request.
+// queryOperation is a GraphQL defaultRequest.
 type (
+	// Operation is a graphQL operation
 	Operation interface {
-		Request() *Req
+		Request() GraphRequest
+		ResponseBodyAs() interface{}
+		IsMutation() bool
+		Name() string
+	}
+	GraphRequest interface {
+		Query() string
 		File(string, string, io.Reader)
 		Files() []File
 		Var(string, interface{})
 		Vars() map[string]interface{}
 		Header(string, string)
 		Headers() http.Header
 	}
-	Request struct {
-		Req *Req
-	}
-	Mutation struct {
-		Req *Req
+	// queryOperation is the regular graphQL query operation
+	queryOperation struct {
```

Comment:
A request could be either a `query` or a `mutation` if struct can be both my suggestion is to name it `defaultOperation`, the field `queryName` could also be just `name`.  wdyt?